assets/blocks/jobs/types.jsx

```diff
@@ -0,0 +1,243 @@
+
+/**
+ * WordPress Dependencies.
+ */
+const { Component } = wp.element,
+	{ __ } = wp.i18n,
+	{ addFilter } = wp.hooks,
+	{
+		withAPIData,
+		CheckboxControl,
+		ToggleControl,
+	} = wp.components;
+
+/**
+ * External Dependencies.
+ */
+import _ from 'lodash';
+
+
+/**
+ * Job Types block attributes.
+ */
+const jobTypesAttributesConfig = {
+	showJobTypeFilters: {
+		type: 'boolean',
+		default: true,
+	},
+	includedJobTypes: {
+		type: 'string', // JSON object
+		default: '{}',
+	},
+	allJobTypes: {
+		type: 'string', // JSON array
+		default: '[]',
+	}
+};
+addFilter(
+	'wpjm_block_jobs_attributes_config',
+	'addJobTypesAttributes',
+	( attributesConfig ) => (
+		Object.assign( attributesConfig, jobTypesAttributesConfig )
+	)
+);
+
+
+/**
+ * Transform a job type attribute value from a JSON-encoded object to a
+ * comma-separated string.
+ *
+ * @param {string} value The JSON-encoded job type attribute value.
+ *
+ * @return {?string} Comma-separated value string for the attribute, or null.
+ */
+function transformJobTypeValue( value ) {
+	let obj = {};
+	try {
+		obj = JSON.parse( value );
+	} catch ( error ) {
+		// On error, just use the empty object.
+	}
+
+	// Get the keys whose values are truthy.
+	let keysToInclude = _.pickBy( obj, ( val ) => val );
+
+	// Create a comma-separated string.
+	value = Object.keys( keysToInclude ).join( ',' );
```

Comment:
Re-assigning arguments is generally frowned upon as it can have other side-effects in certain circumstances. It is generally better to avoid that all together (most linting standards will pick up on this).

Reply:
Yeah, good catch. Done.